src/topology/tietze_extension.lean

```diff
@@ -0,0 +1,383 @@
+      refine ⟨x, hxy.le, _⟩,
+      cases le_total c (g y) with hc hc,
+      { simp [dg0 (or.inr hc), (hg_mem y).2] },
+      { calc g y + dg y ≤ c + (c - a) : add_le_add hc (dgmem _).2
+                    ... = b           : by rw [hsub, add_sub_cancel'_right] } } },
+  /- Now we deal with the case `∀ x, f x ≠ b`. -/
+  choose xl hxl hgb using hg_mem,
+  rcases em (∃ x, f x = b) with ⟨x, rfl⟩|hb',
+  { exact ⟨g, λ y, ⟨xl y, x, hxl y, hgb y⟩, hgf⟩ },
+  have hd : disjoint (range e ∪ g ⁻¹' (Iic c)) (g ⁻¹' {b}),
+  { refine disjoint_union_left.2 ⟨_, disjoint_preimage _ _⟩,
+    { rintro _ ⟨⟨x, rfl⟩, rfl : g (e x) = b⟩,
+      exact hb' ⟨x, (congr_fun hgf x).symm⟩ },
+    { exact set.disjoint_singleton_right.2 hcb.not_le } },
+  rcases exists_bounded_mem_Icc_of_closed_of_le
+    (he.closed_range.union $ is_closed_Iic.preimage g.continuous)
+    (is_closed_singleton.preimage g.continuous) hd (sub_nonneg.2 hcb.le)
+    with ⟨dg, dg0, dgb, dgmem⟩,
+  replace hgf : ∀ x, (g - dg) (e x) = f x,
+  { intro x, simp [dg0 (or.inl $ mem_range_self _), ← hgf] },
+  refine ⟨g - dg, λ y, _, funext hgf⟩,
+  { have hyb : (g - dg) y < b,
+    { rcases (hgb y).eq_or_lt with rfl|hlt,
+      { refine (sub_lt_self_iff _).2 _,
+        calc 0 < g y - c : sub_pos.2 hcb
+           ... = dg y    : (dgb rfl).symm },
+      { exact ((sub_le_self_iff _).2 (dgmem _).1).trans_lt hlt } },
+    rcases hb.exists_between hyb with ⟨_, ⟨xu, rfl⟩, hyxu, hxub⟩,
+    cases lt_or_le c (g y) with hc hc,
+    { rcases em (a ∈ range f) with ⟨x, rfl⟩|ha',
+      { refine ⟨x, xu, _, hyxu.le⟩,
+        calc f x = c - (b - c) : by rw [← hsub, sub_sub_cancel]
+             ... ≤ g y - dg y  : sub_le_sub hc.le (dgmem _).2 },
+      { have hay : a < (g - dg) y,
+        { calc a = c - (b - c)   : by rw [← hsub, sub_sub_cancel]
+             ... < g y - (b - c) : sub_lt_sub_right hc _
+             ... ≤ g y - dg y    : sub_le_sub_left (dgmem _).2 _ },
+        rcases ha.exists_between hay with ⟨_, ⟨x, rfl⟩, ha, hxy⟩,
+        exact ⟨x, xu, hxy.le, hyxu.le⟩ } },
+    { refine ⟨xl y, xu, _, hyxu.le⟩,
+      simp [dg0 (or.inr hc), hxl] } },
+end
+
+/-- **Tietze extension theorem** for real-valued bounded continuous maps, a version for a closed
+embedding. Let `e` be a closed embedding of a nonempty topological space `X` into a normal
+topological space `Y`. Let `f` be a bounded continuous real-valued function on `X`. Let `s` be
+a nonempty convex set of real numbers (we use `ord_connected` instead of `convex` to automatically
+deduce this argument by typeclass search) such that `f x ∈ s` for all `x`. Then there exists
+a bounded continuous real-valued function `g : Y →ᵇ ℝ` such that `g y ∈ s` for all `y` and
+`g ∘ e = f`. -/
+lemma exists_extension_forall_mem_of_closed_embedding (f : X →ᵇ ℝ) {s : set ℝ} {e : X → Y}
+  [hs : ord_connected s] (hf : ∀ x, f x ∈ s) (hne : s.nonempty) (he : closed_embedding e) :
+  ∃ g : Y →ᵇ ℝ, (∀ y, g y ∈ s) ∧ g ∘ e = f :=
```

Comment:
Could you please use `t` for the set of real numbers that contains the range of `f`? In most other statements, `s` is a subset of `Y` and `t` a subset of the reals, so it would be good to have this consistently all over the file. Same thing below.